templates/cluster-template-azure.yaml

```diff
@@ -128,14 +123,11 @@ metadata:
   labels:
     clusterctl.cluster.x-k8s.io/move-hierarchy: "true"
   name: ${CLUSTER_IDENTITY_NAME}
-  namespace: default
 spec:
   type: ServicePrincipal
   clientID: ${AZURE_CLIENT_ID}
   clientSecret:
     name: ${AZURE_CLUSTER_IDENTITY_SECRET_NAME}
     namespace: ${AZURE_CLUSTER_IDENTITY_SECRET_NAMESPACE}
   tenantID: ${AZURE_TENANT_ID}
-  allowedNamespaces:
-    list:
-      - default
+  allowedNamespaces: {}
```

Comment:
I think clusterctl generate should have a `${NAMESPACE}` variable that is the namespace for which the cluster is generated. Please check if it could be of use here.

Reply:
According to [this](https://cluster-api.sigs.k8s.io/clusterctl/commands/generate-cluster.html), we can have `--target-namespace` to define a namespace for the cluster. We can define a `${NAMESPACE}` in our rc file and pass it here if necessary. But why don't we define namespaces in any other of our providers? 
Also, the [allowedNamespaces](https://capz.sigs.k8s.io/topics/multitenancy.html#allowednamespaces) is more specific to identities rather than the cluster.

Reply:
In general, clusterctl will patch the resources to go to the target namespace. So `clusterctl generate -n ns1` will generate resources that deploy into the `ns1` namespace. This is done automatically and nothing is needed on the template itself.

The difference i see here for the Azure provider, is that it needs to know the namespace the cluster is deployed in? Is this assumption correct?

Reply:
No. The namespace is only required to know where the [identity secret](https://capz.sigs.k8s.io/topics/multitenancy.html#identity-types) is stored. The rest of the cluster can be deployed to any namespace akin to the other providers.